channeldb/waitingproof.go

```diff
@@ -223,11 +223,11 @@ func (p *WaitingProof) Encode(w io.Writer) error {
 	return nil
 }
 
-// Decode reads the data from the byte stream and initialize the
+// Decode reads the data from the byte stream and initializes the
 // waiting proof object with it.
 func (p *WaitingProof) Decode(r io.Reader) error {
 	var b [1]byte
-	if _, err := r.Read(b[:]); err != nil {
+	if err := binary.Read(r, byteOrder, b[:]); err != nil {
```

Comment:
We should be able to pass `&p.isRemote` directly into `binary.Read` here, eliminating the declaration of `var b [1]byte` all together.